### ISSUE TITLE
Version 1.9.0 has a "warning: string literal in condition" warning message

### ISSUE THREAD

**Issue by emmahsax** [ruby-git/ruby-git#526]
### Subject of the issue
On the latest new version of Ruby git gem, I see this message:

```
/Users/emmasax/.rbenv/versions/3.0.1/lib/ruby/gems/3.0.0/gems/git-1.9.0/lib/git/lib.rb:663: warning: string literal in condition
```

This error is pointing to this line: https://github.com/ruby-git/ruby-git/blob/07a1167d4706ba4a66ab3025229ca4ed844a934d/lib/git/lib.rb#L663

Looking at the error (StackOverflow describes what the warning message means [here](https://stackoverflow.com/questions/20867709/warning-string-literal-in-condition/20867732)), it seems the easiest way to fix it is to change that line to:

```ruby
arr_opts << '--gpg-sign' if opts[:gpg_sign] && (opts[:gpg_sign] == true || opts[:gpg_sign] == "--gpg-sign=#{opts[:gpg_sign]}")
```

### Your environment
* Ruby git gem 1.9.0
* Ruby version 3.0.1 installed with rbenv

### Steps to reproduce
Using the newest version of the git Ruby gem with Ruby 3.0.1:
```
$ irb
irb(main):002:0> require 'git'
/Users/emmasax/.rbenv/versions/3.0.1/lib/ruby/gems/3.0.0/gems/git-1.9.0/lib/git/lib.rb:663: warning: string literal in condition
=> true
```

### Expected behaviour
```
$ irb
irb(main):002:0> require 'git'
=> true
```

### Actual behaviour
See the `Steps to reproduce` section.

**Comment by jcouball** [comment]
Thank you for the great error report.  I noticed the same warning and added a fix for it in PR #525.  I'll cut a new release soon.

**Comment by jcouball** [comment]
This has been fixed this in git-1.9.1:

```
$ gem install git -v 1.9.1
Fetching git-1.9.1.gem
Successfully installed git-1.9.1
Parsing documentation for git-1.9.1
Installing ri documentation for git-1.9.1
Done installing documentation for git after 0 seconds
1 gem installed
 $ irb
3.0.1 :001 > require 'git'
 => true
3.0.1 :002 >
```

**Comment by emmahsax** [comment]
Thank you @jcouball!
